### PR TITLE
Support for AFC 3v1

### PR DESCRIPTION
**Pull request by kaolpr**
Support for [Advanced FMC Carrier, version 3.1](https://www.ohwr.org/projects/afc/wiki/wiki).

### CODE REVIEW

File: misoc/targets/afc3v1.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    parser.add_argument("--with-spi-flash", action="store_false",
+                        help="enable SPI Flash support ")
+    
+def soc_afc3v1_argdict(args):
+    r = soc_sdram_argdict(args)
+    r["with_spiflash"] = args.with_spi_flash
+    return r
+
+def main():
+
```
Comment (sbourdeauducq) [inline misoc/targets/afc3v1.py]: Remove those blank lines.

File: misoc/targets/afc3v1.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        if with_spiflash:
+            self.add_wb_slave(self.mem_map["spiflash"], 16*1024*1024, self.spiflash.bus)
+
+        if not self.integrated_rom_size:
+            self.flash_boot_address = 0x350000
+            self.register_rom(self.spiflash.bus, 16*1024*1024)
+
+
+class MiniSoC(BaseSoC):
+
```
Comment (sbourdeauducq) [inline misoc/targets/afc3v1.py]: here

File: misoc/targets/afc3v1.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    mem_map = {
+        "ethmac": 0x30000000,  # (shadow @0xb0000000)
+    }
+    mem_map.update(BaseSoC.mem_map)
+
+    def __init__(self, *args, ethmac_nrxslots=2, ethmac_ntxslots=2, **kwargs):
+
```
Comment (sbourdeauducq) [inline misoc/targets/afc3v1.py]: here

File: misoc/cores/sdram_settings.py [hunk 4]
```diff
@@ -239,3 +239,18 @@ class MT41J256M16(SDRAMModule):
...
+    nbanks = 8
+    nrows  = 65536
+    ncols  = 1024
+    # timings (-125 speedgrade)
+    tRP   = 13.75
+    tRCD  = 13.75
+    tWR   = 15
+    tWTR  = 7.5
+    tREFI = 7800
+    tRFC  = 260
```
Comment (sbourdeauducq) [inline misoc/cores/sdram_settings.py]: EOL

**Comment by kaolpr** [comment]
@sbourdeauducq Do you have any further remarks concerning this pull request?

**Comment by sbourdeauducq** [comment]
No.

**Pull request merged**